src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -35,155 +36,191 @@ public void setUp() {
-            ENDPOINT_BEING_TESTED,
-            nonExistentTaskId,
-            new AssignTaskRequest(assigneeId),
-            authenticationHeaders
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.NOT_FOUND.value())
-            .and()
-            .contentType(APPLICATION_JSON_VALUE)
-            .body("timestamp", lessThanOrEqualTo(ZonedDateTime.now().plusSeconds(60)
-                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
-            .body("error", equalTo(HttpStatus.NOT_FOUND.getReasonPhrase()))
-            .body("status", equalTo(HttpStatus.NOT_FOUND.value()))
-            .body("message", equalTo(String.format(
-                LOG_MSG_THERE_WAS_A_PROBLEM_FETCHING_THE_VARIABLES_FOR_TASK,
-                nonExistentTaskId
-            )));
+        boolean isFeatureEnabled = launchDarklyClient.getKey(
+            FeatureFlag.RELEASE_2_ENDPOINTS_FEATURE.getKey(),
+            authenticationHeaders.getValue(AUTHORIZATION));
+
+        if (!isFeatureEnabled) {
+            String nonExistentTaskId = "00000000-0000-0000-0000-000000000000";
+
+            common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+            Response result = restApiActions.post(
+                ENDPOINT_BEING_TESTED,
+                nonExistentTaskId,
+                new AssignTaskRequest(assigneeId),
+                authenticationHeaders
+            );
+
+            result.then().assertThat()
+                .statusCode(HttpStatus.NOT_FOUND.value())
+                .and()
+                .contentType(APPLICATION_JSON_VALUE)
+                .body("timestamp", lessThanOrEqualTo(ZonedDateTime.now().plusSeconds(60)
+                                                         .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+                .body("error", equalTo(HttpStatus.NOT_FOUND.getReasonPhrase()))
+                .body("status", equalTo(HttpStatus.NOT_FOUND.value()))
+                .body("message", equalTo(String.format(
+                    LOG_MSG_THERE_WAS_A_PROBLEM_FETCHING_THE_VARIABLES_FOR_TASK,
+                    nonExistentTaskId
+                )));
+        }
     }
 
     @Test
     public void should_return_a_204_when_assigning_a_task_by_id() {
-        TestVariables testVariables = common.setupTaskAndRetrieveIds();
-        String taskId = testVariables.getTaskId();
+        boolean isFeatureEnabled = launchDarklyClient.getKey(
+            FeatureFlag.RELEASE_2_ENDPOINTS_FEATURE.getKey(),
+            authenticationHeaders.getValue(AUTHORIZATION));
 
-        common.setupOrganisationalRoleAssignment(authenticationHeaders);
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            taskId,
-            new AssignTaskRequest(assigneeId),
-            authenticationHeaders
-        );
+        if (!isFeatureEnabled) {
+            TestVariables testVariables = common.setupTaskAndRetrieveIds();
```

Comment:
Same here this test should apply for both r1 and r2

Reply:
resolved